modules/cudaarithm/src/cuda/threshold.cu

```diff
@@ -101,8 +101,9 @@ double cv::cuda::threshold(InputArray _src, OutputArray _dst, double thresh, dou
 
     const int depth = src.depth();
 
-    CV_Assert( src.channels() == 1 && depth <= CV_64F );
+    CV_Assert( depth <= CV_64F );
     CV_Assert( type <= 4 /*THRESH_TOZERO_INV*/ );
+    src = src.reshape(1);
```

Comment:
Please call this reshape after `dst` mat initialization. `dst` mat should have same number of channels as original input. So call `src = src.reshape(1);` after `GpuMat dst` initialization and call `reshape` for the `dst` mat too.